Microsoft.Azure.Cosmos/src/Query/Core/SqlQuerySpec.cs

```
@@ -43,8 +44,22 @@ public SqlQuerySpec(string queryText, SqlParameterCollection parameters)
         {
             this.QueryText = queryText;
             this.parameters = parameters ?? throw new ArgumentNullException("parameters");
```

Not your change Aditya, but maybe @neildsh knows if this pattern is requirement for some reason here?
This is an internal class. Not sure why we have a pattern of throwing an exception from the constructor instead of either picking the right default or better yet making sure we never pass wrong value. Am I missing something?